airflow/operators/docker_operator.py

```diff
@@ -187,35 +188,28 @@ def execute(self, context):
                 tls=tls_config
             )
 
-        if ':' not in self.image:
-            image = self.image + ':latest'
-        else:
-            image = self.image
```

Comment:
Why were these remove? (I can't work out from the diff alone.

Reply:
Hi @ashb , this is because I found out that this will be handled by `docker` package itself. 

Here this operator is using `docker.APIClient.create_container(image=...)` to create the container under the hood. If the `image` is just the name withOUT tag, say `"fake_image"`, then the program will go search for `"fake_image:latest"` by default (of course it will search for the tag given by user if user has specified the tag).

You can try the code below to validate:
```python
import docker
client = docker.APIClient(base_url='unix://var/run/docker.sock')
client.create_container(image="fake_image")
```

It will tell you something like `"'fake_image:latest' can't be found"` (I don't have Docker on the laptop I'm using at this moment so can't provide the exact exception). Please help double-validate.

Cheers

Reply:
Hey @XD-DENG were you really certain about that? Because I noticed something weird when I use for example just `r-base` as an image name.

The DockerOperator task pulled every image.

```
r-base                                                     latest              62c848eeb175        2 weeks ago         649MB
r-base                                                     3.5.3               62c848eeb175        2 weeks ago         649MB
r-base                                                     3.5.2               880eb7da671b        5 weeks ago         655MB
r-base                                                     3.5.1               bd9edc1a85ed        4 months ago        712MB
r-base                                                     3.5.0               190658892827        9 months ago        692MB
r-base                                                     3.4.4               d1325eaa28ad        11 months ago       667MB
r-base                                                     3.4.3               d1e1c25485af        13 months ago       670MB
r-base                                                     3.4.2               02d3b7e00020        17 months ago       651MB
r-base                                                     3.4.1               0ab131e275e4        19 months ago       1.18GB
r-base                                                     3.4.0               5a6c58403310        22 months ago       656MB
r-base                                                     3.3.3               88436550cddc        2 years ago         635MB
r-base                                                     3.3.2               38270dc4745b        2 years ago         635MB
r-base                                                     3.3.1               7ba1baf9d8bb        2 years ago         657MB
r-base                                                     3.3.0               8f62a54a58ab        2 years ago         970MB
r-base                                                     3.2.5               ee4ea743f431        2 years ago         1.05GB
r-base                                                     3.2.4               9d9c50e41475        3 years ago         1.07GB
r-base                                                     3.2.3               1844ae3ec8d8        3 years ago         1.03GB
r-base                                                     3.2.2               7201dfdf7e21        3 years ago         1.01GB
r-base                                                     3.2.1               62e43f6f9a91        3 years ago         563MB
r-base                                                     3.2.0               9adf6ba7ea84        3 years ago         552MB
r-base                                                     3.1.3               8b37253aa5ea        4 years ago         517MB
r-base                                                     3.1.2               803d5e0e0278        4 years ago         483MB
```

`pip list` show me `docker                        3.7.2`

Reply:
@feluelle let me have a quick check and get back here.

Reply:
@feluelle may you share your log? You may have logs like `Pulling image (xxx) from xxx`. Thanks

Reply:
Hi @feluelle the change was made based on the facts below:

1. For **`pull`**: https://docker-py.readthedocs.io/en/stable/api.html#docker.api.image.ImageApiMixin.pull  In the example, you can find that the `latest` tag is used if no tag is provided together with the repo.

2. For **`create_container`**: you can try to code below

```python
import docker
client = docker.APIClient(base_url='unix://var/run/docker.sock')
client.create_container(image="fake_image")
```
It will give you error "ImageNotFound: 404 Client Error: Not Found ("No such image: `fake_image:latest`")"

Reply:
Sure. https://pastebin.com/raw/NUmS0tYh

Reply:
(That's not all) I think it is for like 3 images